akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -486,9 +486,8 @@ trait PathMatchers {
    * @group pathmatcher
    */
   val JavaUUID: PathMatcher1[UUID] =
-    PathMatcher("""[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}""".r) flatMap { string =>
-      try Some(UUID.fromString(string))
```

Comment:
You removed the try/catch because you now know that all paths matching will not throw in `UUID.fromString`? It seemed a slight bit safer before because we might not know for which input `UUID.fromString` might throw in all JDKs (and also in the future).

Reply:
Yes, I think we're safe here, we know what a UUID should look like. The `IllegalArgumentException` is thrown when the string is too large or when there are no 4 `-`s in the string.

Reply:
Seems reasonable, thanks for the explanation.

Could it then be changed to `.map(UUID.fromString)`?